src/KeyRegistrar.sol

```diff
@@ -0,0 +1,472 @@
+        return avsOperatorKeyInfo[avs][operator][curveType].isRegistered;
+    }
+
+    /**
+     * @notice Initiates key rotation for an operator with a specific AVS
+     * @param avs Address of the AVS
+     * @param operator Address of the operator
+     * @param operatorSetIds Array of operator set IDs to update (only for BN254 keys)
+     * @param curveType Type of curve (ECDSA, BN254)
+     * @param newPubkey New public key bytes
+     */
+    function rotateKey(
+        address avs,
+        address operator,
+        uint32[] calldata operatorSetIds,
+        CurveType curveType,
+        bytes calldata newPubkey
+    ) external onlyAVSRegistrar(avs) {
+        // Check if key is registered
+        if (!avsOperatorKeyInfo[avs][operator][curveType].isRegistered) {
+            revert KeyNotRegistered();
+        }
+
+        // Get old key
+        bytes memory oldKey;
+        if (curveType == CurveType.ECDSA) {
+            oldKey = avsOperatorToECDSAKey[avs][operator];
+            _registerECDSAKey(avs, operator, newPubkey);
+        } else if (curveType == CurveType.BN254) {
+            BN254.G1Point memory oldKeyPoint = avsOperatorToBN254Key[avs][operator];
+            oldKey = abi.encode(oldKeyPoint.X, oldKeyPoint.Y);
+            
+            // Register new key
+            BN254.G1Point memory newKeyPoint = _registerBN254Key(avs, operator, newPubkey);
+            
+            // Update all specified operator sets
+            for (uint256 i = 0; i < operatorSetIds.length; i++) {
+                uint32 operatorSetId = operatorSetIds[i];
+                _updateOperatorSetAggregateBN254Key(avs, operatorSetId, oldKeyPoint, newKeyPoint);
+            }
+        } else {
+            revert InvalidCurveType();
+        }
+
+        // Update rotation timestamp
+        uint256 effectiveBlock = block.number + rotationDelay;
+        avsOperatorKeyInfo[avs][operator][curveType].lastRotationBlock = effectiveBlock;
+
+        emit KeyRotationInitiated(avs, operator, curveType, oldKey, newPubkey, effectiveBlock);
+    }
+
+    /**
+     * @notice Update an operator set's aggregate BN254 key
+     * @param avs Address of the AVS
+     * @param operatorSetId ID of the operator set
+     * @param key BN254 key to add or remove
+     * @param isAddition True to add the key, false to remove it
+     */
+    function _updateAggregateBN254Key(
+        address avs,
+        uint32 operatorSetId,
+        BN254.G1Point memory key,
+        bool isAddition
+    ) internal {
+        BN254.G1Point memory currentApk = avsOperatorSetToAggregateBN254Key[avs][operatorSetId];
+        BN254.G1Point memory newApk;
+        
+        if (isAddition) {
+            newApk = currentApk.plus(key);
+        } else {
+            newApk = currentApk.plus(key.negate());
+        }
+        
+        avsOperatorSetToAggregateBN254Key[avs][operatorSetId] = newApk;
+        
+        emit AggregateBN254KeyUpdated(avs, operatorSetId, newApk);
+    }
```

Comment:
This would be code that would sit outside this contract in the pattern I suggested above , maintained by eigen, and when someone registers a "Key Type" enum/identifier , they would be able to point to an external contract with this handling logic (that would receive a bytes array and decode it to BN254.G1Point, in this case)